src/config/config.rs

```diff
@@ -101,55 +70,73 @@ impl DatabaseSetting {
     } else {
       PgSslMode::Prefer
     };
-    PgConnectOptions::new()
-      .host(&self.host)
-      .username(&self.username)
-      .password(&self.password)
-      .port(self.port)
-      .ssl_mode(ssl_mode)
+    let options = self.pg_conn_opts.clone();
+    options.ssl_mode(ssl_mode)
   }
 
   pub fn with_db(&self) -> PgConnectOptions {
     self.without_db().database(&self.database_name)
   }
-
-  /// Generate a postgresql connection string from the database settings.
-  pub fn to_pg_url(&self) -> String {
-    let ssl_mode = if self.require_ssl {
-      "require"
-    } else {
-      "prefer"
-    };
-    format!(
-      "postgres://{}:{}@{}:{}/{}?sslmode={}",
-      self.username, self.password, self.host, self.port, self.database_name, ssl_mode
-    )
-  }
 }
 
-pub fn get_configuration(app_env: &Environment) -> Result<Config, config::ConfigError> {
-  let base_path = std::env::current_dir().expect("Failed to determine the current directory");
-  let configuration_dir = base_path.join("configuration");
-
-  let builder = InnerConfig::builder()
-        .set_default("default", "1")?
-        .add_source(
-            config::File::from(configuration_dir.join("base"))
-                .required(true)
-                .format(FileFormat::Yaml),
-        )
-        .add_source(
-            config::File::from(configuration_dir.join(app_env.as_str()))
-                .required(true)
-                .format(FileFormat::Yaml),
-        )
-        // Add in settings from environment variables (with a prefix of APP and '__' as
-        // separator) E.g. `APP__APPLICATION__PORT=5001 would set
-        // `Settings.application.port`
-        .add_source(config::Environment::with_prefix("app").separator("__"));
+// Default values favor local development.
+pub fn get_configuration() -> Result<Config, anyhow::Error> {
+  let config = Config {
+    app_env: get_env_var("APPFLOWY_ENVIRONMENT", "local").parse()?,
+    db_settings: DatabaseSetting {
+      pg_conn_opts: PgConnectOptions::from_str(&get_env_var(
+        "APPFLOWY_DATABASE_URL",
+        "postgres://postgres:password@localhost:5433/postgres",
+      ))?,
+      require_ssl: get_env_var("APPFLOWY_DATABASE_REQUIRE_SSL", "false").parse()?,
```

Comment:
how about making these string as const &str

Reply:
you mean the ones wrapped in ""? isn't already static lifetime &str? or are you referring the the result from `get_env_var`?

Reply:
Never mind. It's ok